src/Histogram.hpp

```diff
@@ -43,4 +48,34 @@ class shasta::Histogram : public vector<size_t> {
 };
 
 
+// A histogram class that calculates bin size and finds the appropriate bin to increment given an x value
```

Comment:
Would it make sense to use the [Boost Accumulators library](https://www.boost.org/doc/libs/1_73_0/doc/html/accumulators.html) instead? It has a [`density`](https://www.boost.org/doc/libs/1_73_0/doc/html/accumulators/user_s_guide.html#accumulators.user_s_guide.the_statistical_accumulators_library.density) feature that computes a histogram.

Reply:
ah I didnt know about that. It looks like they have tons of other incremental stats objects